CMakeLists.txt

```diff
@@ -231,6 +232,13 @@ if (HPGMP_ENABLE_CUDA)
     target_link_libraries(hpgmp_device PUBLIC CUDA::cublas CUDA::cusparse CUDA::curand)
     target_link_libraries(hpgmp PUBLIC
                           CUDA::cublas CUDA::cusparse CUDA::curand)
+
+    if(HPGMP_ENABLE_PROFILING)
+      target_compile_definitions(hpgmp PUBLIC HPGMP_WITH_PROFILING)
```

Comment:
Can we move this line out of this HPGMP_ENABLE_CUDA block have a global `if (HPGMP_ENABLE_PROFILING)` block? Then we can have one line that adds the preprocessor define in general.